src/main/java/io/vertx/ext/hawkular/impl/VertxMetricsImpl.java

```diff
@@ -155,8 +157,14 @@ public boolean isEnabled() {
   public void eventBusInitialized(EventBus bus) {
     // Finish setup
     Context context = vertx.getOrCreateContext();
-    sender = new Sender(vertx, options, context);
+    HttpClientOptions httpClientOptions = options.getHttpOptions()
```

Comment:
Please move back the http client setup to the sender. The httpclient must be created in the context. Let's revisit http client sharing later.